dask/array/routines.py

```diff
@@ -312,25 +312,56 @@ def _inner_apply_along_axis(arr,
     )
 
 
-@wraps(np.apply_along_axis)
-def apply_along_axis(func1d, axis, arr, *args, **kwargs):
+def apply_along_axis(func1d, axis, arr, dtype=None, shape=None, *args, **kwargs):
+    """
+    Apply a function to 1-D slices along the given axis. This is
+    a blocked variant of :func:`numpy.apply_along_axis` implemented via
+    :func:`dask.array.map_blocks`
+
+    Parameters
+    __________
+
+    func1d : callable
+        Function to apply to 1-D slices of the array along the given axis
+    axis : int
+        Axis along which func1d will be applied
+    arr : dask array
+        Dask array to which ``func1d`` will be applied
+    dtype : np.dtype, optional
+        The ``dtype`` of the output array. It is recommended to provide this.
+        If not provided, will be inferred by applying the function to a small
+        set of fake data.
+    shape : tuple, optional
+        The ``shape`` of the output array. It is recommended to provide this.
+        If not provided, will be inferred by applying the function to a small
+        set of fake data.
+    args : any
+        Additional arguments to ``func1d``.
+    kwargs : any
+        Additional named arguments to ``func1d``.
+
+    """
     arr = asarray(arr)
 
     # Validate and normalize axis.
-    arr.shape[axis]
```

Comment:
So the purpose of this line is to verify if `axis` is valid and throw otherwise. Could we keep it or would you rather propose something different?

Reply:
I was wondering why that line was there... Yes, it seems fine to keep it, maybe with a slightly more expressive comment

Reply:
Feel free to change the comment as you see fit.